File: Logic Building Problems/Basic Problems/Program to print multiplication table of a number/logic2.js
// printTable() prints table of number and takes
//1 required value that is number of whose teble to be printed
//and an optional input i whose default value is 1

function printTable(n, i = 1) {
    if (i == 11)// base case
        return;
    console.log(n + " * " + i + " = " + n * i);
    i++;
    printTable(n,i);
}

// Driver Code
let n = 5;
printTable(n);

File: Logic Building Problems/Basic Problems/Program to print multiplication table of a number/logic2.cpp
#include <iostream>
using namespace std;

// printTable() prints table of number and takes
// 1 required value that is number of whose teble 
// to be printed and an optional input i whose d
// efault value is 1
void printTable(int n, int i = 1) {
    if (i == 11)
        return;
    cout << n << " * " << i << " = " << n * i << endl;
    i++;
    printTable(n,i);
}

int main() {
    int n = 5;
    printTable(n);
}